courses-app/src/components/Courses/components/CourseCard/CourseCard.tsx

```diff
@@ -0,0 +1,55 @@
+import { FC } from 'react';
+
+import { AuthorType, CourseType } from '../../../../types';
+import { BUTTONS_TEXT, mockedAuthorsList } from '../../../../constants';
+import { formatDuration } from '../../../../helpers/pipeDuration';
+import { formatAuthors } from '../../../../helpers/authorFormatHelper';
+
+import { Button } from '../../../../common/Button/Button';
+
+import {
+	StyledCardWrapper,
+	StyledDataWrapper,
+	StyledInnerWrapper,
+	StyledTextWrapper,
+} from './CourseCard.style';
+
+const CourseCard: FC<CourseType> = (course) => {
+	const { id, title, duration, creationDate, description, authors } = course;
+
+	const findAuthors = (authorsArr: AuthorType[], authors: string[]) => {
+		return authors.map((authorId) => {
+			const author = authorsArr.find(
+				(authorName) => authorName.id === authorId
+			);
+			return author ? author.name : '';
+		});
+	};
+
+	const filteredAuthors = findAuthors(mockedAuthorsList, authors);
```

Comment:
You don't need both variables.  
Just do `filteredAuthors = useMemo(() => {}, [])`
`mockedAuthorsList, authors` are globals in this scope you will be able to use them if you provide the correct depth in the `useMemo`

Reply:
Change